common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -954,8 +956,13 @@ public final ThreadProperties threadProperties() {
         return threadProperties;
     }
 
-    protected boolean wakesUpForTask(@SuppressWarnings("unused") Runnable task) {
-        return true;
+    /**
+     * Marker interface for {@link Runnable} that will not trigger an {@link #wakeup(boolean)} in all cases.
```

Comment:
Should we make it clear that this only holds true as long as the sub-class not overrides `wakesUpForTask` ?